csrc/host_ir/jit.cpp

```diff
@@ -166,27 +180,275 @@ void printLlvmIr(llvm::Function* func, std::string_view msg) {
-// properly, we will support more complex tensor shapes and strides in future
-// PRs.
+llvm::Value* createValueForBinaryOp(
+    BinaryOp* binary_op,
+    std::unordered_map<Val*, llvm::Value*>& val_to_value,
+    llvm::IRBuilder<>& builder) {
+  auto* lhs = binary_op->lhs()->as<Val>();
+  auto* rhs = binary_op->rhs()->as<Val>();
+  llvm::Value* lhs_value =
+      getOrCreateValueForExtent(lhs, val_to_value, builder);
+  llvm::Value* rhs_value =
+      getOrCreateValueForExtent(rhs, val_to_value, builder);
+  if (binary_op->getBinaryOpType() == BinaryOpType::Add) {
+    return builder.CreateAdd(lhs_value, rhs_value);
+  } else if (binary_op->getBinaryOpType() == BinaryOpType::Sub) {
+    return builder.CreateSub(lhs_value, rhs_value);
+  } else if (binary_op->getBinaryOpType() == BinaryOpType::Mul) {
+    return builder.CreateMul(lhs_value, rhs_value);
+  } else if (binary_op->getBinaryOpType() == BinaryOpType::CeilDiv) {
+    // Implement ceilDiv as (a + b - 1) / b
+    llvm::Value* numerator = builder.CreateAdd(lhs_value, rhs_value);
+    llvm::Value* one = builder.getInt64(1);
+    numerator = builder.CreateSub(numerator, one);
+    return builder.CreateUDiv(numerator, rhs_value);
+  }
+  NVF_THROW(
+      "LLVM Lowering Error: Unsupported binary operation type in extent "
+      "calculation: ",
+      binary_op->getBinaryOpType());
+  return nullptr;
+}
+
+llvm::Value* createValueForUnaryOp(
+    UnaryOp* unary_op,
+    std::unordered_map<Val*, llvm::Value*>& val_to_value,
+    llvm::IRBuilder<>& builder) {
+  auto* in = unary_op->in()->as<Val>();
+  llvm::Value* in_value = getOrCreateValueForExtent(in, val_to_value, builder);
+  if (unary_op->getUnaryOpType() == UnaryOpType::Cast) {
+    return in_value;
+  } else if (unary_op->getUnaryOpType() == UnaryOpType::Abs) {
+    llvm::Value* is_negative =
+        builder.CreateICmpSLT(in_value, builder.getInt64(0));
+    llvm::Value* negated = builder.CreateNeg(in_value);
+    return builder.CreateSelect(is_negative, negated, in_value);
+  } else if (unary_op->getUnaryOpType() == UnaryOpType::Neg) {
+    return builder.CreateNeg(in_value);
+  }
+  NVF_THROW(
+      "LLVM Lowering Error: Unsupported unary operation type in extent "
+      "calculation: ",
+      unary_op->getUnaryOpType());
+  return nullptr;
+}
+
+llvm::Value* createValueForExtent(
+    Val* val,
+    std::unordered_map<Val*, llvm::Value*>& val_to_value,
+    llvm::IRBuilder<>& builder) {
+  if (val->isA<IterDomain>()) {
+    if (val->as<IterDomain>()->isBroadcast()) {
+      if (val->as<IterDomain>()->hasExpandedExtent()) {
+        return getOrCreateValueForExtent(
+            val->as<IterDomain>()->expandedExtent(), val_to_value, builder);
+      }
+      return builder.getInt64(1);
+    }
+    return getOrCreateValueForExtent(
+        val->as<IterDomain>()->extent(), val_to_value, builder);
+  } else if (val->isConst()) {
+    return builder.getInt64(val->value().as<int64_t>());
```

Comment:
Simplify this function using early exits. There are too many nested if-elses, making it hard to read.